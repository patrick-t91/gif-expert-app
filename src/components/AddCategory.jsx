import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("One Punch");

  const onInputChange = ({ target }) => {
    // Podemos desestructurar la propiedad target del evento
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange} // Al tener un solo parametro la funcion, no hace falta que lo llamemos aqui, podemos solamente invocar la funcion.
      />
    </form>
  );
};

export default AddCategory;

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}