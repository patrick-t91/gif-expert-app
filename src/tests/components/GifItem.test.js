import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe("Pruebas en GiftItem", () => {
  
  const url =
    "https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=a8630c87ofyabdttjelw7rkl0nyvg50yexkje34mrwau5wns&amp;rid=giphy.gif&amp;ct=g";
  const title = "Hola soy el titulo";

  test("Testing GifGrid: Snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen con el URL y el alt especificado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    /* expect(screen.getByRole('img').src).toBe(url);
    expect(screen.getByRole('img').alt).toBe(title) */
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test('Debe mostrar el titulo en el componente', () => { 
    render( <GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy();
   })
});
