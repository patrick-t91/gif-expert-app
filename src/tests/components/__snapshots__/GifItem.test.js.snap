// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem Testing GifGrid: Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola soy el titulo"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=a8630c87ofyabdttjelw7rkl0nyvg50yexkje34mrwau5wns&amp;rid=giphy.gif&amp;ct=g"
    />
    <p>
      Hola soy el titulo
    </p>
  </div>
</div>
`;
